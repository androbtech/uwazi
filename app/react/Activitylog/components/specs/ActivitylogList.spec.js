import React from 'react';
import { shallow } from 'enzyme';
import Immutable from 'immutable';
import ActivitylogList from '../ActivitylogList.js';

describe('ActivitylogList', () => {
  let props;
  let component;

  const render = () => {
    props = {
      activitylog: Immutable.fromJS([
        { _id: 1, method: 'POST', url: '/api/entities', body: '{"title":"hey"}', query: '{}', time: '2019-06-17T13:36:12.697Z' },
        { _id: 2, method: 'GET', url: '/api/entities', body: '{}', query: '{"_id": "123"}', time: '2019-06-17T13:36:12.697Z' },
        { _id: 3, method: 'DELETE', url: '/api/entities', body: '{"_id":"123"}', query: '{}', time: '2019-06-17T13:36:12.697Z' },
      ])
    };
    component = shallow(<ActivitylogList.WrappedComponent {...props}/>);
  };

  it('should render a table with the activity log entries', () => {
    render();
    expect(component).toMatchSnapshot();
  });
});
