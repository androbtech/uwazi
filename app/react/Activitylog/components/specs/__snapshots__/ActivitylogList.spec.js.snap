// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivitylogList should render a table with the activity log entries 1`] = `
<div
  className="activity-log-list"
>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Method
        </th>
        <th>
          Url
        </th>
        <th>
          User
        </th>
        <th>
          Query
        </th>
        <th>
          Body
        </th>
        <th>
          Time
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="1"
      >
        <td>
          <span
            className="badge btn-color-6"
          >
            POST
          </span>
        </td>
        <td>
          /api/entities
        </td>
        <td>
          -
        </td>
        <td
          className="tdquery"
        >
          {}
        </td>
        <td
          className="tdbody"
        >
          {"title":"hey"}
        </td>
        <td>
          06/17/2019 3:36:12 PM
        </td>
      </tr>
      <tr
        key="2"
      >
        <td>
          <span
            className="badge btn-color-9"
          >
            GET
          </span>
        </td>
        <td>
          /api/entities
        </td>
        <td>
          -
        </td>
        <td
          className="tdquery"
        >
          {"_id": "123"}
        </td>
        <td
          className="tdbody"
        >
          {}
        </td>
        <td>
          06/17/2019 3:36:12 PM
        </td>
      </tr>
      <tr
        key="3"
      >
        <td>
          <span
            className="badge btn-color-2"
          >
            DELETE
          </span>
        </td>
        <td>
          /api/entities
        </td>
        <td>
          -
        </td>
        <td
          className="tdquery"
        >
          {}
        </td>
        <td
          className="tdbody"
        >
          {"_id":"123"}
        </td>
        <td>
          06/17/2019 3:36:12 PM
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
